src/cwe_checker_lib/src/intermediate_representation/expression.rs

```diff
@@ -431,5 +432,79 @@ impl Expression {
     }
 }
 
+impl fmt::Display for Expression {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            Expression::Var(var) => write!(f, "{}", var),
+            Expression::Const(c) => {
+                write!(f, "0x{:016x}:i{}", c, c.bytesize().as_bit_length())
+            }
+            Expression::BinOp { op, lhs, rhs } => match op {
+                BinOpType::IntMult
+                | BinOpType::IntDiv
+                | BinOpType::IntRem
+                | BinOpType::FloatMult
+                | BinOpType::FloatDiv => write!(f, "{} {} {}", lhs, op, rhs),
+                _ => write!(f, "({} {} {})", lhs, op, rhs),
+            },
+            Expression::UnOp { op, arg } => write!(f, "{}({})", op, arg),
+            Expression::Cast { op, size: _, arg } => write!(f, "{}({})", op, arg),
+            Expression::Unknown {
+                description,
+                size: _,
+            } => write!(f, "{}", description),
+            Expression::Subpiece {
+                low_byte,
+                size,
+                arg,
+            } => {
+                if let (Ok(start), Ok(end)) = (u32::try_from(low_byte.0), u32::try_from(size.0)) {
+                    write!(f, "({})[{}₋{}]", arg, start, end)
+                } else {
+                    write!(f, "{}[]", arg)
+                }
+            }
+        }
+    }
+}
+
+impl fmt::Display for BinOpType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            BinOpType::IntEqual => write!(f, "=="),
+            BinOpType::IntNotEqual => write!(f, "!="),
+            BinOpType::IntLess => write!(f, "<"),
+            BinOpType::IntSLess => write!(f, "<"),
+            BinOpType::IntLessEqual => write!(f, "<="),
+            BinOpType::IntSLessEqual => write!(f, "<="),
+            BinOpType::IntAdd => write!(f, "+"),
+            BinOpType::IntSub => write!(f, "-"),
+            BinOpType::IntXOr => write!(f, "^"),
+            BinOpType::IntAnd => write!(f, "&"),
+            BinOpType::IntOr => write!(f, "|"),
+            BinOpType::IntLeft => write!(f, "<<"),
+            BinOpType::IntRight => write!(f, ">>"),
+            BinOpType::IntMult => write!(f, "*"),
+            BinOpType::IntDiv => write!(f, "/"),
+            BinOpType::IntRem => write!(f, "%"),
+            _ => write!(f, "{:?}", self),
+        }
+    }
+}
+
+impl fmt::Display for UnOpType {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            UnOpType::IntNegate => write!(f, "¬"),
+            _ => write!(f, "{:?}", self),
```

Comment:
We should also add the `BoolNegate` operation here. I would actually use `¬` for it, although then we should use something different for `IntNegate`. Maybe `~` (tilde)? This would be the C-operator for bitwise negation.